src/main/java/com/amazon/dlic/auth/http/saml/SamlHTTPMetadataResolver.java

```diff
@@ -15,6 +15,7 @@
 import java.security.AccessController;
```

Comment:
Back when I [migrated the `main` branch](https://github.com/opensearch-project/security/pull/2166) from Apache HttpClient 4 -> Apache HttpClient 5, I needed to retain the dependency on HttpClient 4 because of this class. Is there any version of OpenSAML available with Apache HttpClient 5 so that we can remove this last remaining dependeny on HttpClient 4?

This PR looks good to me.

Reply:
Ahhh I will take a look for sure. Beside found our that `Shibboleth` has its own repository for OpenSAML and the latest version is 4.3.0. I will push changes

Reply:
So 4.3.0 uses HTTP commons version 4. to exclude it completely I could try to prepare PR to the Shibboleth git repo (not sure how it works) or just implement our own soultion with HTTP commons 5 instead of 4. AFAU we need to extend `org.opensaml.saml.metadata.resolver.impl.AbstractReloadingMetadataResolver` class for that. Wdyt?

Reply:
So I pushed 4.3.0 version

Reply:
@cwperks added an issue https://github.com/opensearch-project/security/issues/2932

Reply:
Thank you for filing an issue to track this @willyborankin! From the Shibboleth git, it looks like the HttpMetadataResolver class is using HttpClient 5 on their main branch: https://git.shibboleth.net/view/?p=java-opensaml.git;a=blob;f=opensaml-saml-impl/src/main/java/org/opensaml/saml/metadata/resolver/impl/HTTPMetadataResolver.java;hb=0d8f395fcca7923cb4d4cc5a98730b5f3fca3aa9

I'm looking to see if there is a release schedule to see when its expected that a published version of the jar will have the update.

Reply:
4.3.0 is the way to go. 4.0.1 was published February 11, 2021 and 4.3.0 on January 17, 2023. It looks like they are only publishing new artifacts to the shibboleth repo.